cumulus_library/template_sql/utils.py

```diff
@@ -126,3 +151,52 @@ def is_codeable_concept_array_populated(
     if cursor.fetchone() is None:
         return False
     return True
+
+
+def is_code_populated(
+    schema: str,
+    table: str,
+    base_col: str,
+    cursor,
+    allow_partial: bool = True,
+) -> bool:
+    """Check db to see if a bare code exists and is populated.
+
+    Will execute several exploratory queries to see if the column in question
+    can be queried naively.
+
+    :param schema: The schema/database name
+    :param table: The table to query against
+    :param base_col: the place to start validation from.
+        This can be a nested element, like column.object.code
+    :param cursor: a PEP-249 compliant database cursor
+    :allow_partial: If true, codings which do not have fields expected by the library
+        will still be included, and will need to be manually coerced.
+    :returns: a boolean indicating if valid data is present.
+    """
+
+    # if the source column is missing for some reason (i.e. we're dealing with
+    # conversion to FHIR rather than a true FHIR source and it's incomplete),
+    # we'll return false
```

Comment:
I take slight umbrage with calling the hard-working ETL's i2b2-converted FHIR _incomplete_ and not a _true FHIR source_. :smile:

Reply:
But the FHIR isn't incomplete. It's totes valid spec-compliant FHIR right? The Library is annoyed that it leaves out some optional elements, but that's not a fault of the FHIR.

Reply:
I guess the FHIR itself is valid - the schema we're creating from that FHIR from a SQL perspective causes headaches, but that's a seperate thing. trimmed to something more concise